spec/02-integration/21-request-debug/01-request-debug_spec.lua

```diff
@@ -144,8 +144,7 @@ local function get_output_header(_deployment, path, filter, fake_ip, token)
       ["X-Real-IP"] = fake_ip or "127.0.0.1",
     }
   })
-  assert.not_same(500, res.status)
-  res:read_body() -- discard body
+  assert.not_same(500, res.status, res:read_body())
```

Comment:
does `not_same` take 3 (or more) arguments? I don't fully understand this. Why it is important to discard body, and if so, can we move it above the `assert.not_same`, it that was the intent?

Reply:
> does not_same take 3 (or more) arguments? I don't fully understand this

The third argument is the error message.

> Why it is important to discard body

In this test, it actually doesn't matter since we only send a single HTTP request. Based on my experience, when using the same HTTP client object to send multiple requests, the second request might retrieve the response body of the first request when reading its header.

Reply:
> the second request might retrieve the response body of the first request when reading its header.

Is this behavior of `lua-resty-http`?

Reply:
> > the second request might retrieve the response body of the first request when reading its header.
> 
> Is this behavior of `lua-resty-http`?

@tzssangglass Indeed, it is `lua-resty-http`, but such behavior is unexpected.

Reply:
https://github.com/Kong/kong/blob/b0d5fa2b2a38bcbebeb141093ce1cca467efa740/spec/helpers.lua#L745-L753 I think you're talking about this